packages/website/src/components/config/ConfigTypeScript.tsx

```diff
@@ -1,74 +1,62 @@
+import type { JSONSchema4 } from 'json-schema';
 import React, { useCallback, useEffect, useState } from 'react';
 
-import { shallowEqual } from '../lib/shallowEqual';
-import type { ConfigModel, TSConfig } from '../types';
+import type { PlaygroundSystem, TSConfig } from '../playground/types';
+import { shallowEqual } from '../util/shallowEqual';
 import type { ConfigOptionsType } from './ConfigEditor';
 import ConfigEditor from './ConfigEditor';
-import { getTypescriptOptions, parseTSConfig, toJson } from './utils';
+import { parseTSConfig, schemaToConfigOptions, toJson } from './utils';
 
-interface ConfigTypeScriptProps {
+export interface ConfigTypeScriptProps {
   readonly isOpen: boolean;
-  readonly onClose: (config?: Partial<ConfigModel>) => void;
-  readonly config?: string;
+  readonly onClose: (isOpen: false) => void;
+  readonly system: PlaygroundSystem;
 }
 
-function ConfigTypeScript(props: ConfigTypeScriptProps): JSX.Element {
-  const { onClose: onCloseProps, isOpen, config } = props;
+function readConfigSchema(system: PlaygroundSystem): ConfigOptionsType[] {
+  const schemaFile = system.readFile('/schema/tsconfig.schema');
+  if (schemaFile) {
+    const schema = JSON.parse(schemaFile) as JSONSchema4;
+    if (
+      schema.type === 'object' &&
+      schema.properties?.compilerOptions?.properties
+    ) {
+      return schemaToConfigOptions(
+        schema.properties.compilerOptions.properties,
+      );
+    }
+  }
+  return [];
+}
+
+function ConfigTypeScript({
+  onClose: onCloseProps,
+  isOpen,
+  system,
+}: ConfigTypeScriptProps): JSX.Element {
   const [tsConfigOptions, updateOptions] = useState<ConfigOptionsType[]>([]);
   const [configObject, updateConfigObject] = useState<TSConfig>();
 
   useEffect(() => {
     if (isOpen) {
```

Comment:
_(continuing my comment below on `onClose`)_ Actually, are these `if (isOpen) {` checks still necessary? I can't repro any bugs locally by removing this one.

Reply:
yes, we don't want to update stuff in modal, modal is rendered on page all the time, and if we remvoe it, config will be updated on first render